src/Features/CSharp/Portable/Completion/CompletionProviders/OperatorIndexerCompletionProvider.cs

```diff
@@ -0,0 +1,220 @@
+                _ => null,
+            };
+        }
+
+        internal override async Task<CompletionChange> GetChangeAsync(Document document, CompletionItem item, TextSpan completionListSpan, char? commitKey, bool disallowAddingImports, CancellationToken cancellationToken)
+        {
+            if (item.Properties.TryGetValue(CompletionHandlerPropertyName, out var value))
+            {
+                var completionChange = value switch
+                {
+                    CompletionHandlerConversion => await HandleConversionChangeAsync(document, item, cancellationToken).ConfigureAwait(false),
+                    CompletionHandlerIndexer => await HandleIndexerChangeAsync(document, item, cancellationToken).ConfigureAwait(false),
+                    _ => null,
+                };
+                if (completionChange is { })
+                {
+                    return completionChange;
+                }
+            }
+
+            return await base.GetChangeAsync(document, item, completionListSpan, commitKey, disallowAddingImports, cancellationToken).ConfigureAwait(false);
+        }
+
+        private static async Task<CompletionChange?> HandleConversionChangeAsync(Document document, CompletionItem item, CancellationToken cancellationToken)
+        {
+            var position = SymbolCompletionItem.GetContextPosition(item);
+            if (!item.Properties.TryGetValue(MinimalTypeNamePropertyName, out var typeName))
+            {
+                return null;
+            }
+
+            var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var token = root.FindTokenOnLeftOfPosition(position);
+            // syntax tree manipulations are to complicated if a mixture of conditionals is involved. Some text manipulation is easier here.
+            //                      ↓               | cursor position
+            // white?.Black.White.Black?.White      | current user input
+            // white?.Black.White.Black?.White      | rootExpression (text manipulation starts with this)
+            //       .Black.White                   | parentExpression (needed to calculate the position to insert the closing brace)
+            //                    Black             | identifier at cursor position (gets removed, because the user typed the name of a type)
+            // |----------------------|             | part to replace (TextChange.Span), if identifier is not present: ends at rootExpression.End (after White.)
+            //                   ↑                  | insert closing brace after parentExpression.Span.End
+            // ((Black)white?.Black.White).?.White  | The result. Because we removed the identifier, the remainder after the identifier may be syntactically wrong 
+            //                             ↑        | cursor after the manipulation is placed after the dot
+            var rootExpression = GetRootExpressionOfInvocation(token);
```

Comment:
i would think you'd only have to worry about this case.  if you have a root conditional, then you'd want to replace from teh start of the root, through the `?`.  i would expect you would only do this if it's not *Directly* after `?.` itself.  i.e. if i ahve   `foo?.` typing the `<dot>` there wouldn't offer conversions (though it would offer indexers).

Reply:
Not offering conversions is one possibility here. Another one would be to complete it as `(foo as C)?.`. The `as` operator should also be used in a nullable context if foo can be null. Another related problem is the handling of value types. Given 
`a?.B.$$` and B is `long B { get; }`, the conversion should be `((int?)a?.B).` or `(a?.B as int?).`. Should we *lift* value types to `Nullable<type>` or not? And the other way around: For build-in value types: should we *unlift* (e.g. offering (int?) for type long?)?

We need to decide about the following cases (expression `a` is a placeholder for any MemberAccess (e.g. `d.e[0].f`) and `a?` for any ConditionalAccess (e.g. `d.e[0]?.f`). If we want to also include nullable annotations, more cases need to be considered.

<table>
  <thead>
    <tr>
      <th rowspan="2"></th>
      <th rowspan="2">Target type of conversion (T)</th>
      <th colspan="2">Expression Type</th>
    </tr>
    <tr>
      <th>struct</th>
      <th>class</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td rowspan="2">a.$$</td>
      <td>struct</td>
      <td>((T)a).</td>
      <td>((T)a).</td>
    </tr>
    <tr>
      <td>class</td>
      <td>((T)a).</td>
      <td>((T)a). or (a as T). or (a as T)?.</td>
    </tr>
    <tr>
      <td rowspan=2">a?.$$</td>
      <td>struct</td>
      <td>((T?)a)?. or (a as T?)?. or ((T)a)?.</td>
      <td>((T)a)?. or dont offer conversion</td>
    </tr>
    <tr>
      <td>class</td>
      <td>(a as T)?.</td>
      <td>(a as T)?.</td>
    </tr>
  </tbody>
</table>

Reply:
I added some Nullable<T> lifting support. The tests are here:
https://github.com/dotnet/roslyn/blob/3b4e1aa7ff61500ae4c78f06ddf01c3d4250530e/src/EditorFeatures/CSharpTest/Completion/CompletionProviders/ConversionCompletionProviderTests.cs#L762-L912